src/Artsy/SystemContext.tsx

```diff
@@ -57,11 +66,17 @@ export const SystemContextProvider: SFC<SystemContextProps> = ({
   const relayEnvironment =
     props.relayEnvironment || createRelaySSREnvironment({ user })
 
-  const providerValues = {
-    ...props,
-    user,
-    relayEnvironment,
-  }
+  const [tracking, setTracking] = useState(props.tracking)
+
+  const providerValues = useMemo(() => {
+    return {
+      ...props,
+      relayEnvironment,
+      setTracking,
+      tracking,
+      user,
+    }
+  }, [props.relayEnvironment, props.user, props.tracking])
```

Comment:
why only these three values? Shouldn't any context value change cause cascading updates?

Reply:
These are the only dynamic values that could change and should trigger updates.